pyqtgraph/graphicsItems/LegendItem.py

```diff
@@ -74,13 +78,52 @@ def addItem(self, item, name):
         if isinstance(item, ItemSample):
             sample = item
         else:
-            sample = ItemSample(item)        
-        row = self.layout.rowCount()
+            sample = ItemSample(item)    
         self.items.append((sample, label))
-        self.layout.addItem(sample, row, 0)
-        self.layout.addItem(label, row, 1)
+        self._addItemToLayout(sample, label)
         self.updateSize()
+
+    def _addItemToLayout(self, sample, label):
+        col = self.layout.columnCount()
+        row = self.layout.rowCount()
+        if row:
+            row -= 1
+        nCol = self.columnCount*2
+        #FIRST ROW FULL
+        if col == nCol:
+            for col in range(0,nCol,2):
+                #FIND RIGHT COLUMN
+                if not self.layout.itemAt(row, col):
+                    break
+            if col+2 == nCol:
+                #MAKE NEW ROW
+                col = 0
+                row += 1
+        self.layout.addItem(sample, row, col)
+        self.layout.addItem(label, row, col+1)
+
+    def setColumnCount(self, columnCount):
+        '''
+        change the orientation of all items of the legend 
+        '''
+        if columnCount != self.columnCount:
+            self.columnCount = columnCount
+            self.rowCount = int(len(self.items)/columnCount)
+            for i in range(self.layout.count()-1,-1,-1):
+                self.layout.removeAt(i)  #clear layout
+            for sample, label in self.items:
+                self._addItemToLayout(sample, label) 
+            self.updateSize()
     
+    def getLabel(self, plotItem):
+        """
+        return the labelItem inside the legend for a given plotItem
+        the label-text can be changed via labenItem.setText
+        """
+        for i in self.items:
+            if i[0].item == plotItem:
+                return i[1]
```

Comment:
Just a small complement. The possibility to add a PlotItem several times will no longer be possible when #343 will be merged.